### PR TITLE
[Messenger] Don't deep-merge senders configuration

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #32953 
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Review by fabpot — APPROVED** [review]
Not sure we can do it in 4.3.

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -1389,6 +1391,7 @@ private function addHttpClientSection(ArrayNodeDefinition $rootNode)
                             ->end()
                         ->end()
                         ->arrayNode('scoped_clients')
+                            ->performNoDeepMerging()
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: why not allow deep merge for clients, eg if one wants one more option for the test env?

Comment (fabpot) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: but what if a client wants less options?

Comment (nicolas-grekas) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: it should be still possible, by properly using dev/test/prod + all-envs files,no?

Comment (chalasr) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: if deep merging happens and one needs less options for a given client, then it requires to reset the options to their default values and, if one of those options is scalar and does not have a default value (e.g. `max_redirects`) then you'll get an invalid type exception if you try resetting it with `null`. We would better force copy-pasting I think.

Comment (nicolas-grekas) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: I'm sorry but I don't think this is what we want:
the reported issue happens when array prototypes are found inside array prototypes.
When this happens, we don't want the nested "senders" prototype to be deep merged, I agree with that.
So to me, the `performNoDeepMerging` should be added to the nested prototype, not to its parent.
Same for other options, case by case.
E.g. for http_client, the "query" option should not be deep merged. But the "resolve" one could be, etc.

**Review by sroze — APPROVED** [review]
Makes sense for the Messenger keys indeed. But shouldn't we do this in 4.4 only because it's still clearly a BC break?

**Comment by fabpot** [comment]
I agree that this should be for 4.4.

**Review by nicolas-grekas — CHANGES_REQUESTED** [review]
just putting a "needs work" here, see my last inline comment :)

**Comment by chalasr** [comment]
PR updated to only disable deep merging for the `framework.messenger.routing.senders` prototype and fix the related ticket. Also rebased on 4.4 with CHANGELOG entry.

As @nicolas-grekas suggests, we need to discuss other parts of the config tree case by case, let's move forward on this one.

**Review by lyrixx — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @chalasr.

**Pull request merged**